### ISSUE TITLE
File and folder names are converted to lowercase

### ISSUE THREAD

**Issue by SergioBertolinSG** [owncloud/files_external_dropbox#27]
### Steps to reproduce

0. Enable dropbox app. (release/1.0.1).
1. Set up dropbox mount point, grant access and so on. (all ok).
2. Upload in the dropbox backend several files having uppercase characters.
3. Access dropbox folder using owncloud.

### Expected behaviour
Files keep their case.

### Actual behaviour
All file names are converted to lowercase.

It happens also when uploading files.

### Server configuration

**Operating system**:
Ubuntu 16.04

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
7.0

**Cache**
DB

**ownCloud version:** (see ownCloud admin page)
{"installed":true,"maintenance":false,"needsDbUpgrade":false,"version":"10.0.4.4","versionstring":"10.0.4","edition":"Community","productname":"ownCloud"}

**Updated from an older ownCloud or fresh install:**
Fresh

**The content of config/config.php:**

```

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
Yes, dropbox.

**Are you using encryption:** 
No.

**Logs**
Nothing appears in the logs as warnings or errors.

### Client configuration

**Browser**
Chrome

**Comment by Hemant-Mann** [comment]
@SergioBertolinSG This is a known behaviour

Dropbox file system is case insensitive meaning FOO.txt or foo.txt or Foo.txt are all the same.
I tried to maintain the case while fetching the files/folder list but the Dropbox API was not sending correct response due to which not all the files were visible so I converted all the names to lowercase

**Comment by SergioBertolinSG** [comment]
https://www.dropbox.com/developers/documentation/http/documentation

"Also, while Dropbox is case-insensitive, it makes efforts to be case-preserving. Metadata.name will contain the correct case. Metadata.path_display usually will contain the correct case, but sometimes only in the last path component. If your app needs the correct case for all components, it can get it from the Metadata.name or last path component of each relevant Metadata.path_display entry."

No idea if this may help. 

It would be good to preserve the original filename, knowing that Aaa.txt and AAA.txt would be considered the same file.

**Comment by SergioBertolinSG** [comment]
Changing to critical as it is considered a regression from the previous app.

**Comment by Hemant-Mann** [comment]
In my dropbox app I have a folder **Foo** with this directory structure
```
hemant > ~ > Dropbox > Apps > hemant-oc-share > tree Foo
Foo
├── bar
│   └── baz.txt
└── bar.txt

1 directory, 2 files
```
When I fetch the list of files using dropbox API
```
{
  "entries": [
    {
      ".tag": "folder",
      "name": "Foo",
      "path_lower": "/foo",
      "path_display": "/Foo",
      "id": "id:AhiRpGo73nAAAAAAAAACZg"
    },
    {
      ".tag": "folder",
      "name": "bar",
      "path_lower": "/foo/bar",
      "path_display": "/Foo/bar",
      "id": "id:AhiRpGo73nAAAAAAAAACZw"
    },
    {
      ".tag": "file",
      "name": "bar.txt",
      "path_lower": "/foo/bar.txt",
      "path_display": "/Foo/bar.txt",
      "id": "id:AhiRpGo73nAAAAAAAAACaQ",
      "client_modified": "2018-01-10T08:21:57Z",
      "server_modified": "2018-01-10T08:22:04Z",
      "rev": "61d5531b4dd",
      "size": 12,
      "content_hash": "2b60206c3f87e67012fbd30855e2571156ee504e7681d9344c02f8d6c094068f"
    },
    {
      ".tag": "file",
      "name": "baz.txt",
      "path_lower": "/foo/bar/baz.txt",
      "path_display": "/foo/bar/baz.txt",
      "id": "id:AhiRpGo73nAAAAAAAAACag",
      "client_modified": "2018-01-10T08:22:11Z",
      "server_modified": "2018-01-10T08:22:12Z",
      "rev": "61e5531b4dd",
      "size": 3,
      "content_hash": "b6d544b5784866bc9398f79837f0418c735f9f7786490030bf9923a04f91a7b7"
    }
  ],
  "cursor": "AAE66lc3um56EmkadrLN3xlcc7BbVfZCFvYAa0pdL-4QCf1_mzqzvUYrBIvcEj3mKro6ZSl4HjWeAwk1OtoHhTYft6NiOYafLd3Ztl4UNIFxKhSnKDmIrXORSNsNHoRhYGIwPwqkhy8o946rf6x6ruwAB9y-CaADn5HZ6KIpCbMzeg",
  "has_more": false
}
```

Note that the last file path is being displayed as --> "path_display": "/foo/bar/baz.txt", whereas it should have been "path_display": "/Foo/bar/baz.txt" and **due to this small change in filename this file will not be displayed by Dropbox App** because the underlying Flysystem do some validation with directory listing.

Because when we will open the directory **/Foo/bar** dropbox will give this file **/foo/bar/baz.txt** and flysytem will mark this file as out of scope because `dirname('/foo/bar/baz.txt') != '/Foo/bar'`

Follow these
a) https://github.com/thephpleague/flysystem/blob/master/src/Filesystem.php#L273
b) https://github.com/thephpleague/flysystem/blob/master/src/Util/ContentListingFormatter.php#L43
c) https://github.com/thephpleague/flysystem/blob/master/src/Util/ContentListingFormatter.php#L100
 
Thoughts @PVince81 ?

**Comment by PVince81** [comment]
This listing looks really weird. Why would Dropbox keep the casing for all entries in path_display except for the last entry ? This is inconsistent.

Seems this would require to override flysystem behavior to make it cope with this

**Comment by PVince81** [comment]
From https://www.dropbox.com/developers/documentation/http/documentation

> Also, while Dropbox is case-insensitive, it makes efforts to be case-preserving. Metadata.name will contain the correct case. Metadata.path_display usually will contain the correct case, but sometimes only in the last path component. If your app needs the correct case for all components, it can get it from the Metadata.name or last path component of each relevant Metadata.path_display entry.

**Comment by Hemant-Mann** [comment]
>Seems this would require to override flysystem behavior to make it cope with this

Okay so how do you propose we should do this?

**Comment by PVince81** [comment]
> Okay so how do you propose we should do this?

I'm not familiar enough with Flysystem. Do you think there is a way to bypass or replace Flysystem's default handlers or is that not possible at all ? If not possible at all, might need to ask the Flysystem project for more flexibility :-/

**Comment by Hemant-Mann** [comment]
I think may be we could make some modified classes which extends vendor classes and override the functions like we have done here ?https://github.com/owncloud/files_external_dropbox/blob/master/lib/Storage/Adapter.php#L30

**Comment by PVince81** [comment]
As I understand this was addressed with https://github.com/owncloud/files_external_dropbox/pull/28 and led to new issue https://github.com/owncloud/files_external_dropbox/issues/33 ?

Closing this ticket then, please reopen if I got it wrong

**Comment by Hemant-Mann** [comment]
> As I understand this was addressed with #28 and led to new issue #33 ?
yes this was fixed in #28 

But #33 is an issue in ownCloud itself not with dropbox
